extensions/pkg/controller/utils_test.go

```diff
@@ -163,9 +166,11 @@ var _ = Describe("Utils", func() {
 				},
 			}
 			workerWithAnnotation := worker.DeepCopyObject()
-			ctx := context.TODO()
+			expectedWorker := worker.DeepCopy()
+			delete(expectedWorker.Annotations, annotation)
 
-			c.EXPECT().Patch(ctx, worker, client.MergeFrom(workerWithAnnotation))
+			ctx := context.TODO()
+			test.EXPECTPatch(ctx, c, expectedWorker, workerWithAnnotation)
```

Comment:
> Do you know why with the new version of controller-runtime it now fails with similar error ? I see kubernetes-sigs/controller-runtime#1406 but I cannot correlate it to the failure.

This is explained in the commit message of 0fdab6e6b1d1f858e4bf6e363b66a0dd957b51f1:

> Due to kubernetes-sigs/controller-runtime#1413,
naive gomock expectations don't work anymore. The reason is that the
`mergeFromPatch` struct has now a function attribute (`createPatch`),
and as `gomega` is using `reflect.DeepEqual` under the hood, the
comparison now fails:
>
> Func values are deeply equal if both are nil; otherwise they are not deeply equal.
(from https://golang.org/pkg/reflect/#DeepEqual)
>
> Hence, we introduce our own `test.EXPECTPatch()` function.
>
> Thanks to @timuthy for pointing this out.

I'm also not too happy with it and am very much open for better ideas.

> The reason I am asking is that currently the helper func test.EXPECTPatch is always expecting a MergeFrom PatchOption and I assume this func won't be useful if there is usage of StrategicMergeFrom or anything other PatchOption.

Maybe, but we don't use `StrategicMergeFrom` or any `PatchOption` yet, so the function can be extended as soon as we need it.

Reply:
Ok, thanks for explaining. I guess the proper way to fix this is to adapt the `createPatch` field from func to an interface in the upstream. I can have a look into this one with some lower prio. But yeah, it seems that it is out of the scope of this PR.

Reply:
Yeah, probably @timebertt can help here as the initial author of this change in k-sigs/controller-runtime :)

Reply:
I now opened https://github.com/kubernetes-sigs/controller-runtime/pull/1439. Let me know if you have more elegant solution to the problem in mind.